src/components/SettingsModal.tsx

```diff
@@ -99,6 +99,9 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
       capabilities.dsls.forEach((dsl) => {
         if (dsl.name === value) {
           setLocalSettings({ ...localSettings, dsl: dsl });
+          const tmpIntegration = integrationJson;
+          tmpIntegration.dsl = dsl.name;
```

Comment:
We have to start trying to use the spread operator to prevent mutation, I know I do it some other places, but we're gonna have to try. 😄  Instead of this try:
```
const tmpIntegration = {...integrationJson, dsl: dsl.name}
```